simuleval/agents/pipeline.py

```diff
@@ -75,7 +75,9 @@ def push(
                 upstream_states=upstream_states + states_list[:index],
             )
         self.module_list[-1].push(
-            segment, states[-1], upstream_states=upstream_states + states_list[:index]
+            segment,
+            states[-1],
+            upstream_states=upstream_states + states_list[: len(self.module_list[:-1])],
```

Comment:
just for my understanding, when is `len(self.module_list[:-1])` not equal to `index`?

Reply:
When the len(self.module_list) = 1.
len(self.module_list[:-1]) will be 0 and `index` will not even get assigned a value in this case, resulting in an exception.